### DISCUSSION TITLE
Integrate Stripe with saleor

### DISCUSSION THREAD

**Discussion by codal-ratan** [saleor/saleor#14203] [Q&A]
Hello Saleor community,

I hope this message finds you well. I am currently facing an issue while trying to install an app with Saleor, and I'm seeking some assistance and insights to resolve it.

**Error Message:**

`WARNING saleor.app.tasks Failed to install app. Error: The auth data given during registration request could not be used to fetch app ID. This usually means that App could not connect to Saleor during installation. Saleor URL that App tried to connect: http://localhost:8000/graphql/`

I have encountered the above error message during the installation process, and I'm unsure about the root cause and how to proceed with troubleshooting. It appears to be related to authentication and connectivity.

Here are some details about my environment:

**Saleor Version: "3.15.0-a.0"
App Version: "version":"0.2.0"
Operating System: "mac os"**

**Steps Taken:** 

1. taken Fork on github 
2. run the forked github project in local
3. run the saleor dashboard in local
4. run the saleor graphql python server in local
5. Trying to install stripe app in saleor.
6. getting this error.

I would appreciate it if anyone in the community who has encountered a similar issue or has expertise in Saleor could offer some guidance on how to resolve this error. Any insights, suggestions, or troubleshooting tips would be highly valuable.

Thank you in advance for your assistance. Let's work together to get this issue resolved and help others who might face a similar challenge in the future.

**Comment by slamer59** [comment]
Hi, it seems that it connect to : http://localhost:8000/graphql/
First I would try to change localhost to 127.0.0.1

Are you able to access to http://localhost:8000/graphql/ in your Web Browser ?

**Comment by ghost** [comment]
Hi I am facing the same problem. Similarly, I forked saleor and running it locally. Then I tried installing stripe app from both the command line and GraphQL:
`python .\manage.py install_app https://stripe.saleor.app/api/manifest`
```
mutation InstallApp{
  appInstall(input:{
    manifestUrl: "https://stripe.saleor.app/api/manifest",
    appName: "Stripe",
    permissions: [HANDLE_PAYMENTS],
  }){
    appInstallation{
      status
      id
    }
    errors{
      field
      message
    }
  }
}
```

Is localhost the problem? I can access http://localhost:8000/graphql/ in my web browser.

Any help is greatly appreciated.

**Comment by matteius** [comment]
I'm also getting this issue with a production deployment -- it should be using my DNS url but is instead falling back to localhost:8000 no matter what env vars I try setting.

**Reply by matteius** [reply]
Ok I got this to work finally -- I had to:
Disable DEBUG
Enable CELERY_ALWAYS_EAGER = True (I didn't want to setup a celery worker yet but is required for the add app to work)
add ENABLE_SSL 
and of course use a publicly accessibly saleor API instance.

**Comment by daltonsena** [comment]
@codal-ratan You should take a look at https://docs.saleor.io/docs/3.x/developer/extending/apps/developing-with-tunnels

I think that your problem is related to the tunneling.

**Comment by codal-ratan** [comment]
I'm getting this error 

2023-10-09 12:54:11,163 INFO celery.worker.strategy Task saleor.app.tasks.install_app_task[3e4d1b6a-fa30-4730-84f5-73daa6d06d4e] received [PID:1:MainThread]


saleor-platform-worker-1     | 2023-10-09 12:54:11,897 INFO celery.worker.strategy Task saleor.app.installation_utils.fetch_brand_data_task[0937d14d-b93d-47de-a5be-5cdbcfc03d0e] received [PID:1:MainThread]
saleor-platform-worker-1     | 2023-10-09 12:54:11,937 WARNING saleor.app.tasks Failed to install app. Error: The auth data given during registration request could not be used to fetch app ID. 


saleor-platform-worker-1     |           **This usually means that App could not connect to Saleor during installation. Saleor URL that App tried to connect: http://localhost:8000/graphql/ [PID:23:MainThread]**


saleor-platform-worker-1     | 2023-10-09 12:54:11,941 INFO celery.app.trace Task saleor.app.tasks.install_app_task[3e4d1b6a-fa30-4730-84f5-73daa6d06d4e] succeeded in 0.7760661670001809s: None [PID:23:MainThread]
saleor-platform-worker-1     | 2023-10-09 12:54:12,009 INFO celery.app.trace Task 

saleor.app.installation_utils.fetch_brand_data_task[0937d14d-b93d-47de-a5be-5cdbcfc03d0e] succeeded in 0.1098524169999564s: None [

**Reply by matteius** [reply]
You can't install an app to a backend that is not publicly accessibly I believe, so your issue is the `connect: http://localhost:8000/graphql/`

**Reply by codal-ratan** [reply]
Okay let me try in my publicly accessibly env
